src/PKSim.Core/Model/Simulation.cs

```diff
@@ -45,8 +65,50 @@ protected Simulation() : base(PKSimBuildingBlockType.Simulation)
          ContainerType = ContainerType.Simulation;
          _results = new NullSimulationResults();
          ResultsHaveChanged = false;
+         _compoundPKCache = new Cache<string, CompoundPK>(x => x.CompoundName, x => new CompoundPK());
+      }
+
+      public void ClearPKCache()
+      {
+         _compoundPKCache.Clear();
+      }
+
+
+      public CompoundPK CompoundPKFor(string compoundName)
+      {
+         if (!_compoundPKCache.Contains(compoundName))
+            _compoundPKCache.Add(new CompoundPK { CompoundName = compoundName });
+
+         return _compoundPKCache[compoundName];
+      }
+
+      public void AddCompoundPK(CompoundPK compoundPK)
+      {
+         _compoundPKCache.Add(compoundPK);
       }
 
+      private double? compoundPKValue(string compoundName, Func<CompoundPK, double?> pkValueFunc)
```

Comment:
Why was this moved to the base class?

Reply:
So it is used by both Individual/Population Simulation. There could be one class between Simulation and Population/Individual Simulation but it sounds like an overshooting to me.